src/cli/project/export/conda_explicit_spec.rs

```diff
@@ -0,0 +1,232 @@
+use std::collections::HashSet;
+use std::fs;
+use std::path::{Path, PathBuf};
+
+use clap::Parser;
+
+use crate::cli::cli_config::PrefixUpdateConfig;
+use crate::cli::LockFileUsageArgs;
+use crate::lock_file::UpdateLockFileOptions;
+use crate::Project;
+use rattler_conda_types::{ExplicitEnvironmentEntry, ExplicitEnvironmentSpec, Platform};
+use rattler_lock::{CondaPackage, Environment, Package};
+
+#[derive(Debug, Parser)]
+#[clap(arg_required_else_help = false)]
+pub struct Args {
+    /// Output directory for rendered explicit environment spec files
+    pub output_dir: PathBuf,
+
+    /// Environment to render. Can be repeated for multiple envs. Defaults to all environments
+    #[arg(short, long)]
+    pub environment: Option<Vec<String>>,
+
+    /// The platform to render. Can be repeated for multiple platforms.
+    /// Defaults to all platforms available for selected environments.
+    #[arg(short, long)]
+    pub platform: Option<Vec<Platform>>,
+
+    /// PyPI dependencies are not supported in the conda explicit spec file.
+    /// This flag allows creating the spec file even if PyPI dependencies are present.
+    #[arg(long, default_value = "false")]
+    pub ignore_pypi_errors: bool,
+
+    #[clap(flatten)]
+    pub lock_file_usage: LockFileUsageArgs,
+
+    #[clap(flatten)]
+    pub prefix_update_config: PrefixUpdateConfig,
+}
+
+fn build_explicit_spec<'a>(
+    platform: &Platform,
+    conda_packages: impl IntoIterator<Item = &'a CondaPackage>,
+) -> miette::Result<ExplicitEnvironmentSpec> {
+    let mut packages = Vec::new();
+
+    for cp in conda_packages {
+        let prec = cp.package_record();
+        let mut url = cp.url().to_owned();
+        let hash = prec.md5.ok_or(miette::miette!(
+            "Package {} does not contain an md5 hash",
+            prec.name.as_normalized()
+        ))?;
+
+        url.set_fragment(Some(&format!("{:x}", hash)));
+
+        packages.push(ExplicitEnvironmentEntry {
+            url: url.to_owned(),
+        });
+    }
+
+    Ok(ExplicitEnvironmentSpec {
+        platform: Some(*platform),
+        packages,
+    })
+}
+
+fn render_explicit_spec(
+    target: impl AsRef<Path>,
+    exp_env_spec: &ExplicitEnvironmentSpec,
+) -> miette::Result<()> {
+    if exp_env_spec.packages.is_empty() {
+        return Ok(());
+    }
+
+    let mut environment = String::new();
+    environment.push_str("# Generated by `pixi project export`\n");
+    environment.push_str(exp_env_spec.to_spec_string().as_str());
+
+    fs::write(target, environment)
+        .map_err(|e| miette::miette!("Could not write environment file: {}", e))?;
```

Comment:
Instead of using map_err you can use context to add additional context to the error. This then retains the original error as well.

Reply:
When you say "context" do you mean using `.into_diagnostic().wrap_err_with(|| format!("Could not write environment file: {}", e))?` as shown in https://docs.rs/miette/latest/miette/trait.Context.html?

Reply:
Yep! Either that or `with_context`. I think they do pretty much the same thing.

Reply:
made the change in the latest commit https://github.com/prefix-dev/pixi/pull/1873/commits/c74c69a0e88eb1cb039fe67245e93713ed36ad8f